edx_proctoring/views.py

```diff
@@ -286,6 +286,98 @@ def get(self, request, exam_id=None, course_id=None, content_id=None):  # pylint
         return Response(data)
 
 
+class StudentOnboardingStatusView(ProctoredAPIView):
+    """
+    Endpoint for the StudentOnboardingStatusView
+
+    Supports:
+        HTTP GET: returns the learner's onboarding status relative to the given course_id
+
+    HTTP GET
+        /edx_proctoring/v1/user_onboarding/status?course_id={course_id}&username={username}
+
+    **Query Parameters**
+        * 'course_id': The unique identifier for the course.
+        * 'username': Optional. If not given, the endpoint will return the user's own status.
+            ** In order to view other users' statuses, the user must be course or global staff.
+
+    **Response Values**
+        * 'onboarding_status': String specifying the learner's onboarding status.
+            ** Will return NULL if there are no onboarding attempts, or the given user does not exist
+        * 'onboarding_link': Link to the onboarding exam.
+    """
+    def get(self, request):
+        """
+        HTTP GET handler. Returns the learner's onboarding status.
+        """
+        data = {
+            'onboarding_status': None,
+            'onboarding_link': None
+        }
+
+        attempt_filters = {'proctored_exam__is_practice_exam': True, 'taking_as_proctored': True}
+
+        username = request.GET.get('username')
+        course_id = request.GET.get('course_id')
+
+        if username:
+            # Check that the user is staff if trying to view another user's status
+            if username != request.user.username:
+                if course_id:
```

Comment:
I would prefer to avoid code duplication and have one exit from the condition statement as it is totally equal.

```
if username and username != request.user.username:
    if course_id and not is_user_course_or_global_staff(request.user, course_id) or not request.user.is_staff:
         return Response(
             status=status.HTTP_403_FORBIDDEN,
             data={'detail': _('Must be a Staff User to Perform this request.')
        )
```
Please ignore if a comment doesn't make sense to you.